Add description of lightAdmin and P/DS creation to facility-manager section of help

pwalczysko opened the pull request:
See the https://trello.com/c/Ly5labnO/3-new-role-needs-added-to-help-managing-group-facility-manager-guides-sysadmin-doc - after reading carefully through the ``managing groups`` section and comparing it with ``facility managers`` section, I opted to make the changes rather in ``facility managers``. the ``managing groups`` is aimed at PIs (owners of groups in OMERO), see e.g. ``This section describes how a PI or other owner of an OMERO group.`` from the ``managing groups``. 

Also added description of the new feature in OMERO.web : creating P/D/S for others', see https://github.com/openmicroscopy/openmicroscopy/pull/5402. 
Staged at http://help.staging.openmicroscopy.org/facility-manager.html

pwalczysko commented:
cc @jburel @mtbc @joshmoore

mtbc commented on facility-manager.html:
```diff
@@ -105,10 +106,12 @@
+    privileges. Each level of permission allows different actions to be
     performed on a user's own, or other user's data, and certain
-    administrative tasks.
+    administrative tasks. Administrator with restricted privileges permission levels
```
the naming / possessive grammar aspect of this may need tweaking

pwalczysko commented:
This should fix @mtbc 's comment.

mtbc commented:
That's certainly an improvement, thank you.

jburel commented:
@pwalczysko I am closing this PR for now so it does not get merged by accident until demo is upgraded. Since I do not want to have new doc in the help when we promote the next build to update the version of omero for demo
I will re-open it later on

jburel commented:
Re-opening for 5.4.

hflynn commented on facility-manager.html:
```diff
@@ -105,10 +106,12 @@
-    Owner and Member. Each level of permission allows different actions to be
-    performed on a user's own, or other user's data, and certain
-    administrative tasks.
+    There are three fixed levels of user permission in OMERO, Administrator, Group
```
permission**s**

hflynn commented on facility-manager.html:
```diff
@@ -105,10 +106,12 @@
-    administrative tasks.
+    There are three fixed levels of user permission in OMERO, Administrator, Group
+    Owner and Member as well as one adjustable level, Administrator with restricted
+    privileges (Restricted Administrators). Each level of permission allows different
```
permission**s**

hflynn commented on facility-manager.html:
```diff
@@ -105,10 +106,12 @@
+    There are three fixed levels of user permission in OMERO, Administrator, Group
+    Owner and Member as well as one adjustable level, Administrator with restricted
+    privileges (Restricted Administrators). Each level of permission allows different
+    actions to be performed on a user's own, or other user's data, and certain
```
other -> another

hflynn commented on facility-manager.html:
```diff
@@ -119,6 +122,15 @@
 <p>
+    Administrators with restricted privileges can have varying levels
+    of specific permissions, see <a href="https://www.openmicroscopy.org/latest/omero/sysadmins/admins-with-restricted-privileges.html"
+    target="_blank" class="external-url">Administrators with restricted privileges</a>.
```
I'd add 'sysadmin documentation' here to make clear that they are leaving the help docs and going into the technical docs when they click

hflynn commented on facility-manager.html:
```diff
@@ -119,6 +122,15 @@
+    Administrators with restricted privileges can have varying levels
+    of specific permissions, see <a href="https://www.openmicroscopy.org/latest/omero/sysadmins/admins-with-restricted-privileges.html"
+    target="_blank" class="external-url">Administrators with restricted privileges</a>.
+    Their additional (above Member) specific permissions, such as ability
```
"...such as the ability"

hflynn commented:
Staged at http://help.staging.openmicroscopy.org/facility-manager.html

pwalczysko commented:
@hflynn 

>Do I take it this hasn't been added to the insight admin UI?

Correct.

pwalczysko commented:
The last 2 commits should address the comments of @hflynn

will-moore commented on facility-manager.html:
```diff
@@ -918,10 +980,16 @@
+            the data tree to create a new Project, Dataset or Screen.
+            Right-clicking on Project and creating Dataset will
+            create a Dataset linked to the Project. Link between the
+            Project and Dataset will belong to the selected user too.
```
Might want to mention that you need "Write" privilege etc to be able to create data belonging to other users?

will-moore commented on facility-manager.html:
```diff
@@ -528,6 +541,55 @@
+        <img alt="../images/facilityManagerLightAdminWeb.jpg" class="align-center"
+        src="images/facilityManagerLightAdminWeb.jpg" style="width: 100%;" />
+
+    </li>
```
Maybe mention somewhere that if you are a Restricted Admin you will only be able to create users with similar of fewer privileges (checkboxes will be disabled).

pwalczysko commented:
The last two commits should address @will-moore 's comments.

mtbc commented on facility-manager.html:
```diff
@@ -549,6 +549,12 @@
+   administrators can create restricted administrators. The privileges of the new
+   restricted administrators can be maximally the same as the privileges
+   of the creating restricted administrator. Privileges (checkboxes) which the creator does not
+   posses themselves will be disabled in the OMERO.web Admin tool.</p>
```
possess

will-moore commented on facility-manager.html:
```diff
@@ -549,6 +549,12 @@
+<p>Both administrators with restricted privileges (restricted administrators) and full
+   administrators can create restricted administrators. The privileges of the new
+   restricted administrators can be maximally the same as the privileges
+   of the creating restricted administrator. Privileges (checkboxes) which the creator does not
```
"...of the creating user." Since this could be restricted admin or full admin.

will-moore commented:
Minor comment, but otherwise good to merge.

jburel commented:
In that case I will be more of the opinion to merge since @pwalczysko is away and fix the minor point in a follow-up PR

hflynn commented:
Leaving open for now as we still have a PR for updating Demo to 5.3.3 open and the 5.4 timeline is still up in the air.

mtbc commented:
Once #262 is in then maybe merge `master` in and revert https://github.com/openmicroscopy/ome-help/commit/cffe73bdccb1a05b759a4a4e6d05ef45660e9e09 here.

pwalczysko commented:
The last commit should address the comment of @will-moore https://github.com/openmicroscopy/ome-help/pull/251#pullrequestreview-56041911 and the comment of @mtbc https://github.com/openmicroscopy/ome-help/pull/251#pullrequestreview-56021543.

pwalczysko commented:
Merged ``origin/master`` and removed https://github.com/openmicroscopy/ome-help/commit/cffe73bdccb1a05b759a4a4e6d05ef45660e9e09.

hflynn commented:
Whatever you've done here @pwalczysko I think you should have a merge commit and then the removal of my change and instead you've re-adding all the other changes from the branch...

mtbc commented:
I didn't end up with a merge commit when I tried it locally. Not sure what's going on here.

pwalczysko commented:
I removed again the commits of @hflynn and @jburel  which I have added somehow previously. But not sure about the merging strategy here, no idea what to do @sbesson

hflynn commented:
Let's not worry about it here then, I can open a separate PR to remove my temporary fix

hflynn commented:
Stub reverted in #264 instead

bramalingam commented on facility-manager.html:
```diff
@@ -105,10 +106,12 @@
+    privileges (Restricted Administrators). Each level of permissions allows different
+    actions to be performed on a user's own, or another user's data, and certain
+    administrative tasks. Permission levels of Restricted Administrators
+    are determined at the point of their creation by the creating Administrator.
```
can we call the creating Administrator as the system administrator ?

will-moore replied:
I'm not sure 'system administrator' means anything different than 'Administrator'. But to clarify, we could say: "Restricted Administrators can be created and edited by Administrators or other Restricted Administrators.This includes setting of their permission levels."

bramalingam commented on facility-manager.html:
```diff
@@ -528,6 +541,61 @@
+<p>This requires using the OMERO.web Admin tool.</p>
+
+<p>Both administrators with restricted privileges (restricted administrators) and full
+   administrators can create restricted administrators. The privileges of the new
```
system administrators(administrators with full privileges) instead of full administrators ?

bramalingam commented:
Two very minor comments on the choice of words for system administrators, other than that everything looks fine according to me.

pwalczysko commented:
This should address the comments of Will and Balaji.

mtbc commented on facility-manager.html:
```diff
@@ -545,15 +546,15 @@
-   possess themselves will be disabled in the OMERO.web Admin tool.</p>
+<p>Both Administrators with restricted privileges (Restricted Administrators) and
+   Administrators with full privileges can create Restricted Administrators.
+   The privileges of the new Restricted Administrators can be maximally the same as
```
Just an idea: "can be maximally the same as" could be simply "cannot exceed".

pwalczysko commented:
The last commit addresses Mark's comment.

will-moore commented:
Looks good, thanks.

hflynn commented:
Could probably do with adding mention of the new user role to http://help.openmicroscopy.org/sharing-data.html too and replacing the screenshot for creating new users http://help.openmicroscopy.org/sharing-data.html#admin with the one you've added to the facility managers section

hflynn commented:
Actually, since the sharing data page is focused on groups etc, okay to just reference the facility manager section for creating restricted admins.

I've updated the link in the sysadmin docs on https://github.com/openmicroscopy/ome-documentation/pull/1777

pwalczysko commented:
The last two commits should address [the comment](https://github.com/openmicroscopy/ome-help/pull/251#issuecomment-331434411) of @hflynn

mtbc commented on sharing-data.html:
```diff
@@ -63,6 +63,17 @@
+<p><span class="bold">Restricted Administrators</span> - typically used by Facility Managers, Image Analysts or persons organizing groups and data for other users.
+</p>
+<ul class="bullet-list">
+<li>practically speaking, any specific action can be done by a Restricted Administrator for which they were given corresponding specific permissio. They can perform such action in any group just like Admins (see below).
```
permission

dominikl commented:
Is that deployed somewhere? http://help.staging.openmicroscopy.org/sharing-data.html still seems to show the previous version.

mtbc commented:
Try triggering https://ci.openmicroscopy.org/job/OME-help-staging/.

pwalczysko commented:
@dominikl : I have rerun the build, now staged at http://help.staging.openmicroscopy.org/sharing-data.html

dominikl commented:
Apart from the typo https://github.com/openmicroscopy/ome-help/pull/251#pullrequestreview-65113330 , looks alright (mostly only reviewed last two commits).

pwalczysko commented:
This should fix the typo spotted by @mtbc

hflynn commented on facility-manager.html:
```diff
@@ -120,12 +123,25 @@
     like to any user's data, in any group (except annotate in Private groups)
     and Group Owners can do anything to any user's data in the groups they
     own.
+    <a href="https://docs.openmicroscopy.org/latest/omero/sysadmins/admins-with-restricted-privileges.html" class="external-url">
```
this will become /restricted-admins.html once https://github.com/openmicroscopy/ome-documentation/pull/1783 is merged

hflynn commented on facility-manager.html:
```diff
@@ -120,12 +123,25 @@
+<p>
+    Administrators with restricted privileges can have varying levels
+    of specific permissions, see system administrator documentation
+    section <a href="https://docs.openmicroscopy.org/latest/omero/sysadmins/admins-with-restricted-privileges.html"
```
again, URL update will be needed

hflynn commented on sharing-data.html:
```diff
@@ -63,6 +63,17 @@
+<li>practically speaking, any specific action can be done by a Restricted Administrator for which they were given corresponding specific permission. They can perform such action in any group just like Admins (see below).
+<li>Restricted Administrator can also see and download any user’s data, in any group.
+<li>see system administrator documentation
+section <a href="https://docs.openmicroscopy.org/latest/omero/sysadmins/admins-with-restricted-privileges.html"
```
URL update will be needed here too

pwalczysko commented:
This should address the comments of @hflynn

hflynn commented:
For extra clarity, also on the sharing page, can you remove the 'OMERO system' bit from 'Managing groups as an OMERO system Administrator' and maybe add a line at the top of that section saying that "Administrators in this context can be full system administrators or restricted administrators with the correct privileges".

hflynn commented:
Thanks! That should be the last fix, I'll get this merged as soon as we don't have the 5.3.5 release blocking us

hflynn commented:
Merging now 5.3.5 is out
